src/_common/fireside/post/post-model.ts

```diff
@@ -30,6 +31,47 @@ interface FiresidePostPublishedPlatform {
 	url: string;
 }
 
+export function isInCommunityPinContext(post: FiresidePost, route: Route) {
+	if (post.communities.length > 0) {
+		const communityLink = post.communities[0];
+		const community = communityLink.community;
+		const channelTitle = communityLink.channel!.title;
+		if (
+			route.name === 'communities.view.overview' &&
+			route.params.path === community.path &&
+			route.params.channel === channelTitle
+		) {
+			return true;
+		}
+	}
+	return false;
+}
+
+export function isInPinContext(post: FiresidePost, route: Route) {
```

Comment:
I think the better way to go about it is to return which resource is the pin context based on the route and do no permissions check here. then have another function that checks permissions to pin on a given resource separate from where the pin context is.

Reply:
There are no permission checks in this though?